### PR TITLE
HAAR-1937: refactor POST '/clients/{clientId}/duplicate'

### PR DESCRIPTION
**Pull request by ravmoj**
refactor POST '/clients/{clientId}/duplicate' to 'base-clients/{baseClientId}/clients'

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/integration/ClientsControllerIntTest.kt [hunk 1]
```diff
@@ -496,6 +496,78 @@ class ClientsControllerIntTest : IntegrationTestBase() {
 
       clientRepository.delete(duplicatedClient)
     }
+
+    @Test
+    fun `verify duplicate, updated client`() {
```
Comment (AlTurner-MOJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/integration/ClientsControllerIntTest.kt:501]: Not clear what this test is doing from the description. Each test case should only test one operation i.e. duplicate, not duplicate and update?

Comment (ravmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/integration/ClientsControllerIntTest.kt]: removed unrelated test

**Review by AlTurner-MOJ — DISMISSED** [review]
lgtm

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**